### PR TITLE
Chronos: Add TF2 Forecaster for seq2seq and lstm

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

**Comment by liangs6212** [comment]
related issue: #4023 
list:
- [x]: completed fit/predict/evaluate/save/load method and ut
- [ ]: others API
- [ ]: params description

File: python/chronos/src/bigdl/chronos/forecaster/tf/lstm_forecaster.py [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+from bigdl.chronos.forecaster.tf.base_forecaster import BaseTF2Forecaster
+from bigdl.chronos.model.VanillaLSTM_keras import model_creator
+
+
+class LSTMForecaster(BaseTF2Forecaster):
```
Comment (TheaperDeng) [inline python/chronos/src/bigdl/chronos/forecaster/tf/lstm_forecaster.py:21]: add a example usage as in pytorch LSTMForecaster
https://github.com/intel-analytics/BigDL/blob/315dd847dfde9a52d4733cc51380104a81e43250/python/chronos/src/bigdl/chronos/forecaster/lstm_forecaster.py#L23-L36

File: python/chronos/src/bigdl/chronos/forecaster/tf/lstm_forecaster.py [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+            "loss": loss,
+            "lr": lr,
+            "optim": optimizer,
+        }
+
+        # model creator settings
+        self.model_creator = model_creator
+
+        # distributed settings
+        # self.distributed = distributed
```
Comment (TheaperDeng) [inline python/chronos/src/bigdl/chronos/forecaster/tf/lstm_forecaster.py:87]:
```python
if self.distributed:
    raise NotImplementedError(...)
```

File: python/chronos/src/bigdl/chronos/forecaster/tf/lstm_forecaster.py [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+        # self.distributed_backend = distributed_backend
+        # self.workers_per_node = workers_per_node
+
+        # other settings
+        self.lr = lr
+        self.metrics = metrics
+        self.seed = seed
```
Comment (TheaperDeng) [inline python/chronos/src/bigdl/chronos/forecaster/tf/lstm_forecaster.py:94]: Does `seed` work?

File: python/chronos/src/bigdl/chronos/forecaster/tf/seq2seq_forecaster.py [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+from bigdl.chronos.forecaster.tf.base_forecaster import BaseTF2Forecaster
+from bigdl.chronos.model.Seq2Seq_keras import model_creator
+
+
+class Seq2SeqForecaster(BaseTF2Forecaster):
+
```
Comment (TheaperDeng) [inline python/chronos/src/bigdl/chronos/forecaster/tf/seq2seq_forecaster.py:22]: same here, add a example usage

File: python/chronos/src/bigdl/chronos/forecaster/tf/seq2seq_forecaster.py [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+        self.metrics = metrics
+        self.seed = seed
+
+        # nano setting
+        # current_num_threads = torch.get_num_threads()
+        # self.num_processes = max(1, current_num_threads//8)  # 8 is a magic num
+        # self.onnx_available = True
+        # self.quantize_available = False
+        # self.checkpoint_callback = False
+        super(Seq2SeqForecaster, self).__init__()
```
Comment (TheaperDeng) [inline python/chronos/src/bigdl/chronos/forecaster/tf/seq2seq_forecaster.py:114]: same questions as lstmforecaster

File: python/chronos/test/bigdl/chronos/forecaster/tf/test_lstm_keras_forecaster.py [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+    def test_lstm_forecaster_save_load(self):
+        train_data, test_data = create_data()
+        forecaster = LSTMForecaster(past_seq_len=10,
+                            input_feature_num=10,
+                            output_feature_num=2)
+        forecaster.fit(train_data)
+        with tempfile.TemporaryDirectory() as checkpoint_file:
+            forecaster.save(checkpoint_file)
+            forecaster.load(checkpoint_file)
+        yhat = forecaster.predict(test_data[0], batch_size=32)
```
Comment (TheaperDeng) [inline python/chronos/test/bigdl/chronos/forecaster/tf/test_lstm_keras_forecaster.py:76]: please check the if type of `forecaster.internal` and the predicted value change between the saved model and loaded model

File: python/chronos/src/bigdl/chronos/forecaster/tf/base_forecaster.py [hunk 7]
```diff
@@ -0,0 +1,125 @@
...
+import keras
+
+
+class BaseTF2Forecaster(Forecaster):
+    def __init__(self, **kwargs):
+        self.fitted = False
+        self.internal = self.model_creator({**self.model_config})
+
+    def fit(self, data, validation_data=None, epochs=1,
+            batch_size=32, shuffle=True, verbose='auto'):
```
[the remark this answers is not in the document]
Comment (liangs6212) [reply python/chronos/src/bigdl/chronos/forecaster/tf/base_forecaster.py:28]: ok, I will delete it in the next commit.

File: python/chronos/src/bigdl/chronos/forecaster/tf/base_forecaster.py [hunk 8]
```diff
@@ -0,0 +1,125 @@
...
+        self.internal.save(checkpoint_file)
+
+    def load(self, checkpoint_file, quantize_checkpoint_file=None):
+        """
+        Load the forecaster.
+
+        :params checkpoint_file: The checkpoint file location you want to load the forecaster.
+        :params quantize_checkpoint_file: # TODO will support.
+        """
+        self.internal = keras.models.load_model(checkpoint_file)
```
[the remark this answers is not in the document]
Comment (liangs6212) [reply python/chronos/src/bigdl/chronos/forecaster/tf/base_forecaster.py:124]: yes, Still a `bigdl.nano.tf.keras.Model` after loading, I will add type check in the next commit.

File: python/chronos/src/bigdl/chronos/forecaster/tf/base_forecaster.py [hunk 9]
```diff
@@ -0,0 +1,125 @@
...
+
+from bigdl.chronos.forecaster.abstract import Forecaster
+from bigdl.chronos.metric.forecast_metrics import Evaluator
+import keras
+
+
+class BaseTF2Forecaster(Forecaster):
+    def __init__(self, **kwargs):
+        self.fitted = False
+        self.internal = self.model_creator({**self.model_config})
```
[the remark this answers is not in the document]
Comment (TheaperDeng) [reply python/chronos/src/bigdl/chronos/forecaster/tf/base_forecaster.py:25]: Please keep this format and omit my previous review.

**Review by TheaperDeng — APPROVED** [review]
LGTM, this PR is planned to be merged after #4083 .

**Comment by liangs6212** [comment]
18889: http://10.112.231.51:18889/view/BigDL-PR-Validation/job/BigDL-Chronos-PR-Validation/269/

**Review by TheaperDeng — APPROVED** [review]
LGTM
